manifests/apache.pp

```diff
@@ -1,34 +1,118 @@
+  Stdlib::Port $https_port = 443,
+) {
   $api_path = '/pulp/api/v3'
   $api_url = "http://${pulpcore::api_host}:${pulpcore::api_port}${api_path}"
   $content_path = '/pulp/content'
-  $content_url = "http://${pulpcore::content_host}:${pulpcore::content_port}${content_path}"
-
-  if $pulpcore::manage_apache {
-    include apache
-    apache::vhost { 'pulpcore':
-      servername => $pulpcore::servername,
-      port       => 80,
-      priority   => '10',
-      docroot    => $pulpcore::webserver_static_dir,
-      proxy_pass => [
-        {
-          'path'         => $api_path,
-          'url'          => $api_url,
-          'reverse_urls' => [$api_url],
-        },
-        {
-          'path'         => $content_path,
-          'url'          => $content_url,
-          'reverse_urls' => [$content_url],
-        },
+  $content_base_url = "http://${pulpcore::content_host}:${pulpcore::content_port}"
+  $content_url = "${content_base_url}${content_path}"
+
+  $base_directories = [
+    {
+      provider       => 'directory',
+      path           => $pulpcore::webserver_static_dir,
+      options        => ['Indexes','FollowSymLinks'],
+      allow_override => ['None'],
+    },
+    {
+      'path'            => $content_path,
+      'provider'        => 'location',
+      'request_headers' => [
+        'unset X-CLIENT-CERT',
+        'set X-CLIENT-CERT "%{SSL_CLIENT_CERT}s" env=SSL_CLIENT_CERT',
       ],
+    },
+  ]
+
+  if $pulpcore::remote_user_environ_name {
+    $remote_user_environ_header = $pulpcore::remote_user_environ_name.regsubst(/^HTTP_/, '')
+    $api_directories = [
+      {
+        'path'            => $api_path,
+        'provider'        => 'location',
+        'request_headers' => [
+          "unset ${remote_user_environ_header}",
+          "set ${remote_user_environ_header} \"%{SSL_CLIENT_S_DN_CN}s\" env=SSL_CLIENT_S_DN_CN",
+        ],
+      },
+    ]
+  } else {
+    $api_directories = []
+  }
+
+  $proxy_pass_content = {
+    'path' => $content_path,
+    'url'  => $content_url,
+  }
+
+  $proxy_pass_api = {
+    'path' => $api_path,
+    'url'  => $api_url,
+  }
+
+  case $pulpcore::apache_http_vhost {
+    true: {
+      $http_vhost_name = 'pulpcore'
+
+      include apache
+      include apache::mod::headers
+      apache::vhost { $http_vhost_name:
+        servername  => $pulpcore::servername,
+        port        => $http_port,
+        priority    => $vhost_priority,
+        docroot     => $pulpcore::webserver_static_dir,
+        directories => $base_directories,
+        proxy_pass  => [$proxy_pass_content],
+      }
+
+    }
+    false: {
+      $http_vhost_name = undef
     }
+    default: {
+      $http_vhost_name = $pulpcore::apache_http_vhost
 
-    if $facts['os']['selinux']['enabled'] {
-      selinux::boolean { 'httpd_can_network_connect': }
+      pulpcore::apache::fragment { 'pulpcore':
+        http_content => epp('pulpcore/apache-fragment.epp', {'proxy_pass' => [$proxy_pass_content]}),
```

Comment:
What is `epp` ?

Reply:
https://puppet.com/docs/puppet/latest/lang_template_epp.html

One benefit is that you explicitly pass in scope. This allows using the same fragment twice in the same manifest file. It also becomes clearer what variables are passed in. You can even type check parameters in the template itself rather than relying on it. Until now I've held off on changing all templates but here I want to do the same thing twice with different variables.